lib/factory_bot/attribute_assigner.rb

```diff
@@ -43,7 +43,7 @@ def decorated_evaluator
     end
 
     def methods_invoked_on_evaluator
-      method_tracking_evaluator.__invoked_methods__
+      method_tracking_evaluator.__invoked_methods__ - [:new]
```

Comment:
What are the implications of removing a method from `method_tracking_evaluator.__invoked_methods__`?

To me, this starts to establish a less than ideal pattern of special cases for individual methods. 

Since this is likely a rather uncommon scenario, I wonder if you could work around it using transient attributes and callbacks?

```ruby
    FactoryBot.define do
      factory :post do
        add_attribute(:title) { "Title" }
        add_attribute(:sequence) { "Sequence" }
        transient do 
          new_attribute { true }
        end

        after(:build)  { |post| post.new = new_attribute }
      end
    end
```

Reply:
Hi, @entcheva, thanks for the review!

You are right: it was an ugly fix =) But I think I came up with a better idea.